traceenv.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"context"
+	"os"
+
+	"go.opentelemetry.io/otel/propagation"
+)
+
+const (
+	traceparentHeader = "traceparent"
+	tracestateHeader  = "tracestate"
+)
+
+// see https://github.com/moby/buildkit/pull/2572
+func initOtelContext(ctx context.Context) context.Context {
+	// open-telemetry/opentelemetry-specification#740
+	parent := os.Getenv("TRACEPARENT")
+	state := os.Getenv("TRACESTATE")
+
+	if parent != "" {
```

Comment:
what if `state` is empty?

Reply:
It's ignored: https://github.com/open-telemetry/opentelemetry-go/blob/v1.1.0/propagation/trace_context.go#L142-L145